saskatoon/sitebase/views.py

```diff
@@ -36,8 +38,14 @@ def dispatch(self, request, *args, **kwargs):
         """Redirect new pickleaders"""
 
         user = self.request.user
-        if user.is_authenticated and user.is_onboarding:
+
+        if user.is_authenticated:
+            if not user.password_set:
+                return redirect('change_password')
```

Comment:
shouldn't this check come after the `is_onboarding` ? iiiuc an onboarding pickleader has `password_set = False`